trapper/commands.py

```diff
@@ -94,6 +103,90 @@ def add_subparser(
         return subparser
 
 
+@Subcommand.register("distributed-run")
+class DistributedRun(Subcommand):
+    """trapper's run command that enables running an experiment in
+    a distributed way.
+
+    Usage:
+        Basic:
+            ` trapper distributed-run experiment.jsonnet <--other-kwargs vals>`
+    """
+
+    def add_subparser(
+        self, parser: argparse._SubParsersAction
+    ) -> argparse.ArgumentParser:
+        description = """Distributed run the experiment specified in the config file which
+         specify the training and/or evaluation of a model on a dataset."""
+        subparser = parser.add_parser(
+            self.name,
+            description=description,
+            help="Start distributed training of a model.",
+        )
+
+        subparser.add_argument(
+            "config_path",
+            type=str,
+            help="path to the experiment config file in `json` or `jsonnet`"
+            " format describing the model, dataset and other details.",
+        )
+
+        subparser.set_defaults(func=torch_distributed_run)
+
+        return subparser
+
+
+def run_distributed(args):
+    """
+    Taken from https://github.com/pytorch/pytorch/blob/master/torch/distributed/run.py
+    and modified to properly run by trapper.
+    """
+    if args.standalone:
+        args.rdzv_backend = "c10d"
+        args.rdzv_endpoint = "localhost:29400"
+        args.rdzv_id = str(uuid.uuid4())
+        log.info(
+            f"\n**************************************\n"
+            f"Rendezvous info:\n"
+            f"--rdzv_backend={args.rdzv_backend} "
+            f"--rdzv_endpoint={args.rdzv_endpoint} "
+            f"--rdzv_id={args.rdzv_id}\n"
+            f"**************************************\n"
+        )
+
+    config, cmd, cmd_args = torch_config_from_args(args)
+
+    # Monkey-patching the original args extracted from torch's
+    # config parser. Effectively, running the following cmd
+    # $ torchrun TORCHRUN_OPTS /path/to/trapper/__main__.py run TRAPPER_OPTS
+    #
+    training_script_idx = cmd_args.index("distributed-run")
+    trapper_main_script = str(PROJECT_ROOT / "trapper/__main__.py")
+    list_replace(cmd_args, trapper_main_script, training_script_idx)
+    cmd_args.insert(training_script_idx + 1, "run")
+
+    elastic_launch(
+        config=config,
+        entrypoint=cmd,
+    )(*cmd_args)
+
+
+def torch_distributed_parse_args():
+    torch_parser = torch_distributed_args_parser()
+    dist_args, _ = torch_parser.parse_known_args()
+    for flag in dist_args.training_script_args[1:]:
+        opt, f, v = torch_parser._parse_optional(flag)
+        if opt is not None:
+            setattr(dist_args, opt.dest, v)
+            dist_args.training_script_args.remove(flag)
+    return dist_args
+
+
+@record
```

Comment:
Might an explanation here, possibly a link etc to explain the need for this function and-or its record wrapper.